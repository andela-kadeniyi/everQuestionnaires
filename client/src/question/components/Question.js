import React from 'react';
import PropTypes from 'prop-types';

const Question = ({ name, label }) => (
  <div>
    <div>
      {name}
    </div>
    <div>
      {label}
    </div>
  </div>  
);

Question.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired
};

export default Question;