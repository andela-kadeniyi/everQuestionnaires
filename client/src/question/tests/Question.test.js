import React from 'react';
import { shallow } from 'enzyme';
import Question from '../components/Question';
import { formattedData } from './testData';


const setup = (question = formattedData) => {
  const component = shallow(
    <Question {...question}  />
  );

  return {
    component
  };
}

describe('Question component', () => {
  it('should render name and label', () => {
    const { component } = setup();
    expect(component.text()).toContain(formattedData.name)
    expect(component.text()).toContain(formattedData.label)
  });
});
