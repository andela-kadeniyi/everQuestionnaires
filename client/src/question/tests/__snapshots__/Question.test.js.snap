// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question component renders self with question props  1`] = `
<div>
  <div>
    Ragdoll traits
  </div>
  <div>
    describe ragdoll traits
  </div>
</div>
`;
